codec/src/main/java/io/netty/handler/codec/LengthFieldBasedFrameDecoder.java

```diff
@@ -439,10 +442,17 @@ protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception
         int actualFrameLength = frameLengthInt - initialBytesToStrip;
         ByteBuf frame = extractFrame(ctx, in, readerIndex, actualFrameLength);
         in.readerIndex(readerIndex + actualFrameLength);
-        frameLengthInt = -1; // start processing the next frame
+        resetFrameLengthInt(); // start processing the next frame
         return frame;
     }
 
+    /**
+     * allow repeat read frame
+     */
+    public void resetFrameLengthInt() {
```

Comment:
Should this API be public? What is the motivation to allow this to be called externally?

Reply:
> Should this API be public? What is the motivation to allow this to be called externally?

done ,  do this